Add PostgreSQL as a possible viewer

cevian opened the pull request:
Add documentation for how to use PostgreSQL with pgai to access the dataset.

julien-c commented on docs/source/_toctree.yml:
```diff
@@ -44,6 +44,8 @@
         - local: polars
           title: Polars
+        - local: postgresql
+          title: PostgreSQL
```
can you also add the link inside `parquet_process.md`?

julien-c commented on docs/source/postgresql.md:
````diff
@@ -0,0 +1,53 @@
+in a streaming fashion.
+
+```sql
+select ai.load_dataset('squad');
````
```suggestion
select ai.load_dataset('rajpurkar/squad');
```

does it also work with namespaced names? (all datasets on HF are namespaced now)

julien-c reviewed (commented):
cool addition! 

could make sense to add it to https://huggingface.co/docs/hub/datasets-libraries too in the future, maybe

cevian commented:
@julien-c addressed your comments. Yeah https://huggingface.co/docs/hub/datasets-libraries would be cool. Do you think we should try to get Push to hub abilities in before we do that?

julien-c commented:
Push to hub would be awesome! do you see some demand for this from `pgai` users?

cevian commented:
Not yet. But we've just added load_datasets so I think it will take people a bit to catch on.

julien-c reviewed (approved):
ok pinging @lhoestq for a review but this PR lgtm

HuggingFaceDocBuilderDev commented:
The docs for this PR live [here](https://moon-ci-docs.huggingface.co/docs/dataset-viewer/pr_3121). All of your documentation changes will be reflected on that endpoint. The docs are available until 30 days after the last update.

lhoestq reviewed (approved):
Great ! Also interested in a push to hub feature, but not need for it to start sharing about this online imo !

Let me know if you post about it online I can re-share with the community :)

Btw I've seen people interested in RAG applications for code documentation lately, let me know if you ever do a demo on this topic

> could make sense to add it to https://huggingface.co/docs/hub/datasets-libraries too in the future, maybe

We might re-organize this indeed, it's a bit suprising to me to have libraries in the Viewer's docs compared to libraries integrated with the Hub directly (especially for tools like `pgai` that provide deep integrations). Anyway I'll keep you posted if we move this page to make it more visible :)

cevian commented:
@lhoestq we shared this here: [X post](https://x.com/avthars/status/1869842858834079916) and on [linkedIn](https://www.linkedin.com/posts/avthars_postgresql-huggingface-opensource-activity-7275609557299523584-3_Vq?utm_source=share&utm_medium=member_desktop) and on [bsky](https://bsky.app/profile/ceviantech.bsky.social/post/3ldorraoppc27). Thanks for re-sharing,

I'll let you know about demos involving code documentation for sure.

lhoestq commented:
Awesome ! glad to see it's getting good reception as well :) 

> I'll let you know about demos involving code documentation for sure.

ok !
